lib/note-detail.jsx

```diff
@@ -34,22 +30,34 @@ export default React.createClass( {
 		return note && note.id;
 	},
 
-	componentWillReceiveProps: function( { note } ) {
-		this.setState( {
-			content: get( note, 'data.content', '' )
-		} );
+	componentWillReceiveProps: function() {
+		const { note } = this.props;
+
+		if ( ! this.isValidNote( note ) || ! this.noteEditor ) return;
+
+		if ( saveTimerId ) clearTimeout( saveTimerId );
+
+		// Save note if content changes haven't been saved yet
+		if ( note.data.content !== this.noteEditor.value ) {
+			this.props.onChangeContent( note, this.noteEditor.value );
+			analytics.tracks.recordEvent( 'editor_note_edited' );
+		}
```

Comment:
This was my attempt to save the note content before a change happens, or if the user switches notes. So if you start typing, then immediately switch notes before the timer fires.

I will test that server change scenario, I think there might be content loss in that case!